src/hotspot/cpu/aarch64/macroAssembler_aarch64.cpp

```diff
@@ -4428,6 +4428,22 @@ void MacroAssembler::load_klass(Register dst, Register src) {
   }
 }
 
+void MacroAssembler::restore_cpu_control_state_after_jni(Register tmp1, Register tmp2) {
+  if (RestoreMXCSROnJNICalls) {
+    Label OK;
+    get_fpcr(tmp1);
+    mov(tmp2, tmp1);
+    // Set FPCR to the state we need. We do want Round to Nearest. We
+    // don't want non-IEEE rounding modes or floating-point traps.
+    bfi(tmp1, zr, 22, 4); // Clear DN, FZ, and Rmode
+    bfi(tmp1, zr, 8, 5);  // Clear exception-control bits (8-12)
+    eor(tmp1, tmp1, tmp2);
```

Comment:
Hmm? So . . .
1) We ensure tmp1 has the bits we want by clearing DN FZ Rmode and Exception bits
2) we XOR tmp1 with the original bits (saved in tmp2) and put the 'difference' in tmp1 (?)
3) If the difference is zero we skip
4) Otherwise we write tmp1 (i.e. the 'difference' bits) to fpcr (???)

Should this not be
```
    get_fpcr(tmp1);
    mov(tmp2, tmp1);
    bfi(tmp1, zr, 22, 4);
    bfi(tmp1, zr, 8, 5);
    eor(tmp2, tmp1, tmp2)
    cbz(tmp2, OK);
    set_fpcr(tmp1);
    bind(OK);
```

Reply:
Argh, yes. The curse of the last-minute change...

Good catch.